.github/workflows/revert-testing.yml

```diff
@@ -0,0 +1,40 @@
+name: 'Revert "[TEST TMP]" commit'
+on:
+  push:
+    branches: [master]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "build"
+  build:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 2
+
+      - name: 'Revert "[TEST TMP]" commit'
+        run: |
+          set -euxo pipefail;
+
+          git config --global user.email "github.action@ci-artifacts.psap"
+          git config --global user.name "Github Action"
+
+          COMMIT_MSG=$(git log --pretty='format:%Creset%s' --no-merges -1)
+          COMMIT_AUTHOR=$(git show -s --format='%an <%ae>' HEAD)
+
+          if ! echo "$COMMIT_MSG" | grep '^\[TEST TMP\]' -q; then
+            echo "Not a temporary test commit, do nothing."
+          fi
+
+          echo "Temporary test commit detected, reverting $COMMIT_MSG ..."
```

Comment:
Any reason not to simply `git reset --hard HEAD~`?

Reply:
yes, `reset` removes the commit, while `revert` keeps it (both of them) in the history

I like the idea of keeping the trace of how the PR was tested as part of the history

but result is identical indeed

Reply:
Yeah keeping it definitely makes sense